[2025-06-18T12:59:12.965+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-06-18T12:59:12.996+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: usdpkr_forecasting_modular.evaluate_stacked manual__2025-06-18T12:58:43.512936+00:00 [queued]>
[2025-06-18T12:59:13.007+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: usdpkr_forecasting_modular.evaluate_stacked manual__2025-06-18T12:58:43.512936+00:00 [queued]>
[2025-06-18T12:59:13.007+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 3
[2025-06-18T12:59:13.024+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): evaluate_stacked> on 2025-06-18 12:58:43.512936+00:00
[2025-06-18T12:59:13.030+0000] {standard_task_runner.py:63} INFO - Started process 6762 to run task
[2025-06-18T12:59:13.032+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'usdpkr_forecasting_modular', 'evaluate_stacked', 'manual__2025-06-18T12:58:43.512936+00:00', '--job-id', '713', '--raw', '--subdir', 'DAGS_FOLDER/usdpkr_forecasting_dag.py', '--cfg-path', '/tmp/tmpk37eyzhv']
[2025-06-18T12:59:13.035+0000] {standard_task_runner.py:91} INFO - Job 713: Subtask evaluate_stacked
[2025-06-18T12:59:13.078+0000] {task_command.py:426} INFO - Running <TaskInstance: usdpkr_forecasting_modular.evaluate_stacked manual__2025-06-18T12:58:43.512936+00:00 [running]> on host 9c4806d77c92
[2025-06-18T12:59:13.177+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='usdpkr_forecasting_modular' AIRFLOW_CTX_TASK_ID='evaluate_stacked' AIRFLOW_CTX_EXECUTION_DATE='2025-06-18T12:58:43.512936+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-06-18T12:58:43.512936+00:00'
[2025-06-18T12:59:13.178+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-06-18T12:59:13.189+0000] {logging_mixin.py:188} INFO - Loading Stacked model from /opt/***/data/models/stacked_model.pkl
[2025-06-18T12:59:13.190+0000] {logging_mixin.py:188} INFO - Loading selected features from /opt/***/data/models/selected_features.pkl
[2025-06-18T12:59:13.190+0000] {logging_mixin.py:188} INFO - Loading train/test data from /opt/***/data/models/train_test_data.pkl
[2025-06-18T12:59:13.248+0000] {logging_mixin.py:188} INFO - Error in evaluate_stacked: 'X_test_selected'
[2025-06-18T12:59:13.249+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-06-18T12:59:13.249+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 400, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/inflation_modules/model_steps.py", line 1025, in evaluate_stacked
    X_test_selected = selected_data['X_test_selected']
                      ~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^
KeyError: 'X_test_selected'
[2025-06-18T12:59:13.262+0000] {taskinstance.py:1206} INFO - Marking task as UP_FOR_RETRY. dag_id=usdpkr_forecasting_modular, task_id=evaluate_stacked, run_id=manual__2025-06-18T12:58:43.512936+00:00, execution_date=20250618T125843, start_date=20250618T125912, end_date=20250618T125913
[2025-06-18T12:59:13.273+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 713 for task evaluate_stacked ('X_test_selected'; 6762)
[2025-06-18T12:59:13.286+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2025-06-18T12:59:13.306+0000] {taskinstance.py:3498} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-06-18T12:59:13.309+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
